### PR TITLE
code-refactored

### PR DESCRIPTION
**Pull request by NeerajGartia21**
I have done some refactoring in the code to have the code.

### CODE REVIEW

File: lib/codec.js [hunk 1]
```diff
@@ -100,6 +122,11 @@
   var longTooBig = f.validator('must be less than 2^53 (javascript safe integer limitation)',
                                f.lessThan(Math.pow(2, 53)));
   var stringOrNullCheck = f.validator('must be a String or null', stringOrNull);
+  var bufferCheck = f.validator('must be a buffer',isBuffer);
+
+  function isBuffer(x){
```
Comment (rigazilla) [inline lib/codec.js:127]: is this used only above? If yes I would avoid a new function

File: lib/codec.js [hunk 2]
```diff
@@ -368,4 +395,19 @@
     }
   }
 
+  var WrappedMessage = (function() {
```
Comment (rigazilla) [inline lib/codec.js:398]: Can we move `WrappedMessage` inside `encodeWrappedMessage` since it's only used there?

**Review by rigazilla — COMMENTED** [review]
Approach looks good, but testsuite is failing, probably some more work is needed.
Plus I've added some comments.
```
# in terminal 1
15:48 $ SERVER_VERSION=13.0.10.Final ./run-servers.sh

#in terminal 2
15:40 $ node --trace-deprecation node_modules/jasmine-node/lib/jasmine-node/cli.js spec/protostream_spec.js  --captureExceptions --forceexit --junitreport
.F..TypeError: Cannot read properties of undefined (reading 'decode')
    at decodeWrappedMessage (/home/rigazilla/git/js-client/lib/protocols.js:794:49)
    at /home/rigazilla/git/js-client/node_modules/underscore/underscore-node-f.cjs:1192:30
    at /home/rigazilla/git/js-client/lib/functional.js:75:29
    at /home/rigazilla/git/js-client/lib/functional.js:93:22
    at reducer (/home/rigazilla/git/js-client/node_modules/underscore/underscore-node-f.cjs:1361:14)
    at Function.<anonymous> (/home/rigazilla/git/js-client/node_modules/underscore/underscore-node-f.cjs:1368:12)
    at /home/rigazilla/git/js-client/lib/functional.js:92:28
    at decodeObject (/home/rigazilla/git/js-client/lib/protocols.js:326:19)
    at /home/rigazilla/git/js-client/lib/protocols.js:364:18
    at Protocol29.decodeBody (/home/rigazilla/git/js-client/lib/protocols.js:357:15)
```

File: lib/codec.js (hunk 1 above)
Comment (NeerajGartia21) [reply lib/codec.js:127]: Not used now but I think it will be used when we will encode basic data types

File: lib/codec.js (hunk 2 above)
Comment (NeerajGartia21) [reply lib/codec.js:398]: Yeah, but I think it will also be used in the decodeWrappedMessage function.

Comment (rigazilla) [reply lib/codec.js:398]: Ok. Just a detail: I would have avoided the definition of `encode` and `decode` functions. Maybe returning `wrappedMessage` directly?

Comment (NeerajGartia21) [reply lib/codec.js:398]: Yes sir. That would be great imo also. I will change that.

File: lib/codec.js [hunk 3]
```diff
@@ -88,6 +93,29 @@
...
+
+  function encodeProtobufInstance(root){
+    return function(obj){
+      return root.encode(obj).finish();
+    }
+  }
+
+  function encodeProtobuf(message,typeId){
+    var encodedMessage = encodeProtobufInstance(message.$type)(message);
+    var messageTypeId = typeId.apply(null,[message.$type.fullName]);
```
Comment (rigazilla) [inline lib/codec.js:108]: Is this preferable to `typeId(message.$type.fullName)` ?

Comment (NeerajGartia21) [reply lib/codec.js:108]: I don't have such preference. Just wanted to do this in fp so written in that way.
Should I change that?

Comment (rigazilla) [reply lib/codec.js:108]: Why do you consider `.apply`more fp oriented than calling `typeId` directly?
I would say that a direct call to `typeId` is more readable.

Comment (NeerajGartia21) [reply lib/codec.js:108]: Ok sir. I will change that.

**Comment by rigazilla** [comment]
merged thanks @NeerajGartia21 !

**Pull request merged**